tdms/include/fieldsample.h

```diff
@@ -0,0 +1,43 @@
+/**
+ * @file fieldsample.h
+ * @brief Class that stores the extracted field values at user-defined vertices.
+ */
+#pragma once
+
+#include "arrays.h"
+#include "field.h"
+#include "simulation_parameters.h"
+
+class FieldSample{
+
+private:
+  double**** tensor = nullptr;
+
+public:
+  mxArray* mx;       // Matlab array
+
+  Vector<int> i;     // Indices along the x-direction of locations at which to sample the field
+  Vector<int> j;     // Indices along the y-direction of locations at which to sample the field
+  Vector<int> k;     // Indices along the z-direction of locations at which to sample the field
+  Vector<double> n;  // Vector of the moments of the field to sample
+
+  explicit FieldSample(const mxArray *ptr);
+
+  // Return true if all vectors in this instance are non-empty (have size > 0)
```

Comment:
```suggestion
  /** Return true if all vectors in this instance are non-empty (have size > 0) */
```
(Keep doxygen happy)